scripts/fdbshow

```diff
@@ -0,0 +1,192 @@
+        if not br_port_str:
+            return
+        
+        offset = len("ASIC_STATE:SAI_OBJECT_TYPE_BRIDGE_PORT:")
+        oid_pfx = len("oid:0x")
+        for s in br_port_str:
+            br_port_id = s[(offset + oid_pfx):]
+            ent = self.db.get_all('ASIC_DB', s, blocking=True)
+            port_id = ent[b"SAI_BRIDGE_PORT_ATTR_PORT_ID"][oid_pfx:]
+            self.if_br_oid_map[br_port_id] = port_id 
+
+
+    def fetch_fdb_data(self):
+        """
+            Fetch FDB entries from ASIC DB. 
+            FDB entries are sorted on "VlanID" and stored as a list of tuples
+        """
+        self.db.connect(self.db.ASIC_DB)
+        self.bridge_mac_list = []
+        
+        fdb_str = self.db.keys('ASIC_DB', "ASIC_STATE:SAI_OBJECT_TYPE_FDB_ENTRY:*")
+        if not fdb_str:
+            return
+
+        oid_pfx = len("oid:0x")
+        for s in fdb_str:
+            fdb_entry = s.decode()
+            try:
+                fdb = json.loads(fdb_entry .split(":", 2)[-1])
+            except ValueError as e: 
+                print e.message
+                continue
+           
+            ent = self.db.get_all('ASIC_DB', s, blocking=True)
+            br_port_id = ent[b"SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID"][oid_pfx:]
+            try:
+                port_id = self.if_br_oid_map[br_port_id]
+                if_name = self.if_oid_map[port_id]
+            except KeyError as e:
+                continue
+            
+            self.bridge_mac_list.append((int(fdb["vlan"]),) + (fdb["mac"],) + (if_name,))
+
+        self.bridge_mac_list.sort(key = lambda x: x[0])
+        return
+    
+    
+    def get_iter_index(self, key_value=0, pos=0):
+        """
+            Get the starting index of matched entry
+        """
+        if pos == 0:
+            self.bridge_mac_list.sort(key = lambda x: x[pos])
+        keys = [r[pos] for r in self.bridge_mac_list]
+        try:
+            return keys.index(key_value)
+        except IndexError as e:
+            print e.message
+
+
+    def display_specific(self, port, vlan):
+        """
+            Display the FDB entries for specified vlan/port.
+        """
+        output = []
+
+        if vlan is not None and port is None:
+            vlan = int(vlan)
+            s_index = self.get_iter_index(vlan)
+            fdb_list = [fdb for fdb in self.bridge_mac_list[s_index:] if fdb[0] == vlan]
+
+        elif vlan is None and port is not None:
+            self.bridge_mac_list = natsorted(self.bridge_mac_list, key = lambda x: x[2])
+            s_index = self.get_iter_index(port, 2)
+            fdb_list = [fdb for fdb in self.bridge_mac_list[s_index:] if fdb[2] == port]
+
+        else:
+            vlan = int(vlan)
+            s_index = self.get_iter_index(vlan)
+            fdb_list = [fdb for fdb in self.bridge_mac_list[s_index:]
+                        if fdb[0] == vlan and fdb[2] == port]
```

Comment:
Simply the conditions into one?

Reply:
Addressed.

Reply:
It is not simplified, and become more complex...

---
In reply to: [140849540](https://github.com/Azure/sonic-utilities/pull/106#discussion_r140849540) [](ancestors = 140849540)

Reply:
I think it now handles the three conditions -  
1 - All arguments None
2 - Some arguments None
3 - None of the arguments are None. 
Could you suggest a different way?

Reply:
Agree its slightly complex since I accounted for adding future options. Revisited the logic and made it simple. Thanks for pointing out.